App.js

```diff
@@ -1,9 +1,363 @@
+          <TouchableHighlight onPress={() => pressPlay()}>
+            <Text style={styles.itemText}>{item.name}</Text>
+          </TouchableHighlight>
+        </View>
+      );
+    }
+    if (item.key === 7) {
+      return (
+        <View style={styles.item}>
+          <Text style={styles.itemText}>{score}</Text>
+        </View>
+      );
+    }
+    return (
+      <View style={styles.item}>
+        <Text style={styles.itemText}>{item.name}</Text>
+      </View>
+    );
+  };
+
+  // render board
+  const renderBoard = ({ item }) => {
+    if (item.invisible === true) {
+      return <View style={[styles.item, styles.itemInvisible]} />;
+    }
+    // Show ROLL button
+    if (item.name === 'ROLL') {
+      return (
+        <View style={styles.item}>
+          <TouchableHighlight onPress={() => pressRoll(item.key)}>
+            <Text style={styles.itemText}>{item.name}</Text>
+          </TouchableHighlight>
+        </View>
+      );
+    }
+    // Create Even button and mark it if it has been selected
+    if (item.name === 'EVEN') {
+      return (
+        <View style={even ? [styles.item, styles.markSpot] : styles.item}>
+          <TouchableHighlight onPress={() => pressEven()}>
+            <Text style={styles.itemText}>{item.name}</Text>
+          </TouchableHighlight>
+        </View>
+      );
+    }
+    // Create Odd button and mark it if it has been selected
+    if (item.name === 'ODD') {
+      return (
+        <View style={odd ? [styles.item, styles.markSpot] : styles.item}>
+          <TouchableHighlight onPress={() => pressOdd()}>
+            <Text style={styles.itemText}>{item.name}</Text>
+          </TouchableHighlight>
+        </View>
+      );
+    }
+    // Show ROLL NUMBER
+    if (item.key === 11) {
+      return (
+        <View style={styles.item}>
+          <Text style={styles.itemText}>{roll}</Text>
+        </View>
+      );
+    }
+    // mark current spot on board
+    if (position === item.boardNumber && (item.invisible === undefined || item.invisible === false)) {
+      return (
+        <View style={[styles.item, styles.markSpot]}>
+          <Text style={styles.itemText}></Text>
+        </View>
+      );
+    }
+    return (
+      <View style={styles.item}>
+        <Text style={styles.itemText}>{item.name}</Text>
+      </View>
+    );
+  };
+
+  // press Play button
+  const pressPlay = useCallback(() => {
+    setGameState(() => {
+      return {
+        roll: null,
+        even: false,
+        odd: false,
+        position: 0,
+        message: 'Roll again',
+        optMessage: 'Kim',
+        score: 0,
+        board: JSON.parse(JSON.stringify(newBoard)),
+      };
+    });
+  }, []);
+
+  // press Even button
+  const pressEven = useCallback(() => {
+    setGameState(prevGameState => {
+      let workEven;
+      let workMessage = 'Roll again';
+      let workOptMessage = 'Kim';
+      if (prevGameState.even === true) {
+        workEven = false;
```

Comment:
So if you re-hit even does it toggle odd on? That seems a little odd, I would expect if I hit even and even is already selected it just stays even

Reply:
oh, or does can you have both even and odd not selected, then this makes more sense

Reply:
Yes.....you can decide to select neither odd/even.....then the board works the old way without any risk/gain.